### PR TITLE
balloon_hotplug.with-migrate

### PR DESCRIPTION
**Pull request by ybduan**
balloon_hotplug.with-migrate: Add migration test after hotplugging/unplugging balloon device

ID: 1467186

Signed-off-by: yduan <yduan@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
(1/3) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.balloon_hotplug: PASS (174.91 s)
 (2/3) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.balloon_hotplug.with-migrate.migrate_after_hotplug: PASS (198.81 s)
 (3/3) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.balloon_hotplug.with-migrate.migrate_after_unplug: PASS (189.18 s)

**Comment by huangyum** [comment]
Please rebase this PR.

File: qemu/tests/cfg/balloon_hotplug.cfg [hunk 1]
```diff
@@ -3,14 +3,27 @@
     type = balloon_hotplug
     monitor_type = qmp
     monitors = qmp1
-    balloon_repeats = 100
     virtio_balloon_pause = 10.0
     free_mem_cmd = cat /proc/meminfo |grep MemFree
+    run_sub_test_after_balloon = no
```
Comment (huangyum) [inline qemu/tests/cfg/balloon_hotplug.cfg]: I prefer to move this line to L29.

File: qemu/tests/balloon_hotplug.py [hunk 2]
```diff
@@ -63,8 +68,14 @@ def run(test, params, env):
             err += ("\nDevice is still in qtree %ss after unplug:\n%s"
                     % (pause, vm.monitor.info("qtree")))
 
+        if params.get("migrate_after_unplug", "no") == "yes":
+            del vm.params["balloon"]
+            del vm.params["balloon_dev_devid"]
+            del vm.params["balloon_dev_add_bus"]
+            vm.migrate(float(params.get("mig_timeout", "3600")))
```
Comment (huangyum) [inline qemu/tests/balloon_hotplug.py]: Would 3600s be too long?

**Comment by ybduan** [comment]
Close this pull request and will send a new patch.

**Pull request closed without merging**